pdfrw/objects/pdfname.py

```diff
@@ -5,15 +5,31 @@
 from .pdfobject import PdfObject
 
 
+
 class PdfName(object):
     ''' PdfName is a simple way to get a PDF name from a string:
 
                 PdfName.FooBar == PdfObject('/FooBar')
     '''
+
+    whitespace = '\x00 \t\f'
+    delimiters = r'()<>{}[\]/%'
+    forbidden = whitespace + delimiters
+
     def __getattr__(self, name):
         return self(name)
 
     def __call__(self, name, PdfObject=PdfObject):
-        return PdfObject('/' + name)
+        obj = PdfObject('/' + name)
+        # whitespace and delimiters in the name object should be encoded
+        if any((c in self.forbidden) for c in name[1:]):
```

Comment:
I think we should optimize this -- this is exactly what re.compile('(your|list|of|items|here)').split() is good for -- it will push the search for characters that need to be encoded down into the regex C engine where they will be more efficient.  You can precompile the compile and the split function, and can also precompile a dictionary that maps each character to its encoded value.  Then the code to do this would look something like:
<pre>
    split_by_bad = split(obj)
    if len(split_by_bad) != 1:
        split_by_bad[1::2] = (lookup_replacements[x] for x in split_by_bad[1::2])
        obj.encoded = ''.join(split_by_bad)

</pre>